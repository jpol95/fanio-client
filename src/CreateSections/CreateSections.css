.submit-sections-button{
    display: block;
    margin: auto;
}

.submit-sections-button:disabled, .submit-installments-button:disabled, .submit-fandom-button:disabled,
 .submit-review-button:disabled{
    background-color: white;
    color:rgb(102, 97, 97);
    cursor: default
}