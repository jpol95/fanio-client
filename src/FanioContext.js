import React from 'react'

const FanioContext = React.createContext({
    fandomList: [],
    reviewList: [],
    sectionList: [], 
    subList: [],
    tagList: [],
    reviewTagList: [],
    installmentList: [],
    handleSubmitReview: () => {}, 
    handleAddFandom: () => {}, 
    handleSubmitInstallments: () => {}, 
    handleGetReview: () => {}
})

export default FanioContext