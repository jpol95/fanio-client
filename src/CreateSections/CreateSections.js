import React from "react";
import CreateSingleSection from "../CreateSingleSection/CreateSingleSection";
import FanioContext from '../FanioContext'
import './CreateSections.css'
import typeList from '../type-list'
import FetchService from "../FetchService";

export default class CreateSections extends React.Component {

  static contextType = FanioContext

  getType = () => {
    return typeList[this.getInstallment().type]
  }

  //fix classes to not use type from context, then on with debugging the get calls for the seeded data

  getInstallment = () => {
      return this.context.installmentList.find(installment => Number(this.props.match.params.installmentId) === installment.id)
  }

  genListName = this.props.match.params.sectionId ? "subList" : "sectionList"
  listName = this.props.match.params.sectionId ? this.getType().subName : this.getType().sectionName
  parentName = this.props.match.params.sectionId ? "section": "installment"


  state = {
    numSections: 0,
    sectionList: [],
    // title: "",
    // sectionList: []
  };

  handleNumSections = (e) => {
    const sectionList = [];
    const numSections = e.target.value;
    for (let i = 0; i < numSections; i++) {
      sectionList.push({});
    }
    this.setState({
      numSections,
      sectionList
    });
  };

  createSectionForms() {
    const sectionFormArray = []
    for (let i = 0; i < this.state.numSections; i++) {
      sectionFormArray.push(<CreateSingleSection {...this.props} type={this.getType()} handleAddSection={this.handleAddSection} sectionId={i} />);
    }
    return sectionFormArray
  }

  handleAddSection = (section) => {
    // console.log(this.props.match.params[`sectionId`])
    const sectionListCopy = [...this.state.sectionList];
    sectionListCopy[section.sectionId] = {
      ...section,
      [`${this.parentName}Id`]: Number(this.props.match.params[`${this.parentName}Id`])
    };
    // console.log(sectionListCopy)
    this.setState({
      ...this.state, sectionList: sectionListCopy,
    });
  };

  handleSubmitSections = async (e) => {
    e.preventDefault()
    console.log("kjvenkejnekrjnfejk")
    const link = this.parentName === "installment" ? `/sections/section/${this.props.match.params.installmentId}` : `/sections/sub/${this.props.match.params.sectionId}`
    const sectionCopy = [...this.state.sectionList]
    const sections = await FetchService.postSections(sectionCopy, link)
    this.props.history.push(`/users/1/fandoms/${this.props.match.params.fandomId}/installment-view/${this.props.match.params.installmentId}`) //CHANGE
    this.context.handleSubmitSections(sections, this.genListName)
  }

//is there a way to combine the add seasons and add episodes form?
//figure out why form is not submitting properly

  render() {
    // console.log("creating Section forms") 
    //HOW IS THIS.PROPS.INSTALLID NOT THROWING AN ERROR IM PRETTY SURE IM NOT PASSING ANY PROPS INTO THIS CLASS
    return (
      <form className="add-sections" onSubmit={this.handleSubmitSections}>
        <label htmlFor={`section-title-${this.props.installId}`}> 
          How many {this.listName}s would you like to add?
        </label>
        <input
          id={`section-title-${this.props.installId}`}
          type="number"
          onChange={this.handleNumSections}
        />
        {this.createSectionForms()}
        <button className="submit-sections-button" type="submit">Add {this.listName}s </button>
      </form>
    );
  }
}
