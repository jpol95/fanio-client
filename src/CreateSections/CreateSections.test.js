import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import CreateSections from "./CreateSections";
import FanioContext from "../FanioContext";

describe("Create Sections component", () => {
  const props = {
    match: {
      params: {
        userId: 1,
        installmentId: 1,
      },
    },
    installId: 1,
  };
  const installmentList = [
    { id: 1, title: "Doctor Who TV Series", type: "Show", fandomId: 1 },
    { id: 2, title: "Parks and Rec TV Series", type: "Show", fandomId: 3 },
    { id: 3, title: "Buffy TV Series", type: "Show", fandomId: 4 },
    { id: 4, title: "Buffy Comic Series", type: "Comic series", fandomId: 4 },
    { id: 5, title: "Supernatural TV Series", type: "Show", fandomId: 2 },
    { id: 6, title: "Harry Potter Books", type: "Book series", fandomId: 5 },
  ];
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <FanioContext.Provider value={{ installmentList: installmentList }}>
        {" "}
        <CreateSections {...props} />{" "}
      </FanioContext.Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <FanioContext.Provider value={{ installmentList: installmentList }}>
          {" "}
          <CreateSections {...props} />{" "}
        </FanioContext.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
