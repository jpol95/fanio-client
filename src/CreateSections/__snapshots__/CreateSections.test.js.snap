// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Installment component renders the UI as expected 1`] = `
Array [
  " ",
  <form
    className="add-sections"
    onSubmit={[Function]}
  >
    <label
      htmlFor="section-title-1"
    >
      How many 
      season
      s would you like to add?
    </label>
    <input
      id="section-title-1"
      onChange={[Function]}
      type="number"
    />
    <button
      className="submit-sections-button"
      type="submit"
    >
      Add 
      season
      s 
    </button>
  </form>,
  " ",
]
`;
